### PR TITLE
Fix opal_check_pmi.m4 for newer PMIx

### PR DESCRIPTION
**Pull request by rhc54**
Quick fix: provide paragraph for PMIx v4.x

Signed-off-by: Ralph H Castain <rhc@open-mpi.org>

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Can one of the admins verify this patch?

**Comment by jdinan** [comment]
ok to test

**Comment by davidozog** [comment]
I just verified this does fix #812.  Thanks @rhc54!

In the past we've preferred to take entire upstream OPAL files as opposed to individual patches (to avoid future merges/maintenance on our end).  What do you think in this case @jdinan?

**Comment by rhc54** [comment]
Ah, didn't know that - you are welcome to do so, if you like. This was taken from the upstream file.

File: config/opal_check_pmi.m4 [hunk 1]
```diff
@@ -303,6 +303,19 @@ AC_DEFUN([OPAL_CHECK_PMIX],[
 
            # if it does exist, then we need to parse it to find
            # the actual release series
+           AS_IF([test "$opal_external_pmix_version_found" = "0"],
+                 [AC_MSG_CHECKING([version 4x])
+                  AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+                                                      #include <pmix_version.h>
+                                                      #if (PMIX_VERSION_MAJOR < 4L)
```
Comment (jdinan) [inline config/opal_check_pmi.m4]: Is it better to write `PMIX_VERSION_MAJOR != 4L` so we don't need to update this for PMIx 5?

**Comment by jdinan** [comment]
I'm fine with this patch for now.  We can file an issue to sync with upstream later.  🤷‍♂️

File: config/opal_check_pmi.m4 (hunk 1 above)
Comment (rhc54) [reply config/opal_check_pmi.m4]: Actually, we chose this method so it won't have to be changed - we are checking that the version is 4 or higher. If it is, then all the code that uses PMIx 4 or lower APIs will still be good. If you someday update to PMIx 5 APIs, then you would need to check for 5 or higher.

Comment (jdinan) [reply config/opal_check_pmi.m4]: Ok, as long as you guys promise to never break backward compatibility.  :neckbeard: 🤞

Comment (rhc54) [reply config/opal_check_pmi.m4]: Yeah, that is the promise we made!

Comment (davidozog) [reply config/opal_check_pmi.m4]: We'll be just fine as long as nobody retires!  😉

**Pull request merged**